python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -53,33 +66,54 @@ def __init__(self):
         ret_val_doc += "For the 'update_blocked' status, data will contain a reason key containing an explanation."
 
         self.parameters = {"return_value": {"description": ret_val_doc, "type": "dict" }}
-            
+
+    def _parse_arguments(self, parameters):
+        """
+        Parses the list of arguments from the command line.
+
+        :param parameters: The content of argv that hasn't been processed by the tank command.
+
+        :returns: The core version. None if --version wasn't specified.
+        """
+        parser = TkOptParse()
```

Comment:
As I said, I'd rather wait to have a better idea of the direction we'd want to take with OptionParser in general with Toolkit.